WordPress/gutenberg
# Block Editor: Introduce block locking UI

## Description
Part of #29864.

PR introduced block locking options modal as described in specs here - https://github.com/WordPress/gutenberg/issues/29864#issuecomment-1049037410. Users can access it via the Block Settings dropdown menu.

**Note:** I omitted the "Edit" locking option. It is because it's not supported by API yet, and I think its rollout should be more granular - maybe we can start with Reusable and Template Part blocks.

### Questions
* Lock icon is only visible when block movement is locked. Should I display it for other lock states as well?
* When parent block has `templateLock` attribute set, should lock icon appear for child blocks?
* Do we want to support locking multi-selected blocks? The APIs support this, but I wasn't sure about UI/UX.

### Todos

## Testing Instructions
1. Open a Post or Page.
2. Insert any block.
3. Find block lock settings within the ellipses dropdown menu.
4. Confirm that a block movement or removal can be locked/unlocked. 

## Screenshots 

https://user-images.githubusercontent.com/240569/156547833-5d780d80-f412-4c14-b3af-d878f804aa8a.mp4

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
